src/monitord/monitord.c

```diff
@@ -87,60 +181,86 @@ void Monitord()
         }
 #endif
 
-        /* Day changed, deal with log files */
-        if (today != p->tm_mday) {
-            if (mond.rotate_log) {
-                sleep(mond.day_wait);
-                /* Daily rotation and compression of ossec.log/ossec.json */
-                w_rotate_log(mond.compress, mond.keep_log_days, 1, 0, mond.daily_rotations);
-            }
-
-            /* Generate reports */
-            generate_reports(today, thismonth, thisyear, p);
-            manage_files(today, thismonth, thisyear);
-
-            today = p->tm_mday;
-            thismonth = p->tm_mon;
-            thisyear = p->tm_year + 1900;
-        } else if (mond.rotate_log && mond.size_rotate > 0) {
-            if (stat(path, &buf) == 0) {
-                size = buf.st_size;
-                /* If log file reachs maximum size, rotate ossec.log */
-                if ( (unsigned long) size >= mond.size_rotate) {
-                    w_rotate_log(mond.compress, mond.keep_log_days, 0, 0, mond.daily_rotations);
+        if (mond.enabled && mond.rotation_enabled) {
+            /* Calculate the logs size only if rotation by size is active */
+            if (mond.min_size > 0 || mond.max_size > 0) {
+                if (stat(path_ossec, &buf) < 0) {
+                    merror("Couldn't stat '%s' file due to '%s'", path_ossec, strerror(errno));
+                } else {
+                    size = buf.st_size;
+                }
+                if (stat(path_ossec_json, &buf_json) < 0) {
+                    merror("Couldn't stat '%s' file due to '%s'", path_ossec_json, strerror(errno));
+                } else {
+                    size_json = buf_json.st_size;
                 }
             }
 
-            if (stat(path_json, &buf) == 0) {
-                size = buf.st_size;
-                /* If log file reachs maximum size, rotate ossec.json */
-                if ( (unsigned long) size >= mond.size_rotate) {
-                    w_rotate_log(mond.compress, mond.keep_log_days, 0, 1, mond.daily_rotations);
+            /* Rotation by size (min_size) and interval */
+            if (mond.min_size > 0 && mond.interval > 0) {
```

Comment:
This code is also duplicated. If it is possible, consider to move it to a common place.

Reply:
It's not possible due to crossed dependencies. A rework of significant Wazuh code would be needed in order to achieve this.